game/hiscore.cc

```diff
@@ -1,36 +1,40 @@
 #include "hiscore.hh"
 
+#include "configuration.hh"
 #include "libxml++-impl.hh"
 
+
 #include <algorithm>
 #include <sstream>
 #include <stdexcept>
 
-bool Hiscore::reachedHiscore(unsigned score, unsigned songid, std::string const& track) const {
+bool Hiscore::reachedHiscore(unsigned score, unsigned songid, unsigned level, std::string const& track) const {
 	if (score > 10000) throw std::logic_error("Invalid score value");
 	if (score < 2000) return false; // come on, did you even try to sing?
 
 	unsigned position = 0;
 	for (auto const& elem: m_hiscore) {
 		if (elem.songid != songid) continue;
 		if (elem.track != track) continue;
+		if (elem.level != level) continue;
 		if (score > elem.score) return true; // seems like you are in top 3!
 		if (++position == 3) return false; // not in top 3 -> leave
 	}
 	return true; // nothing found for that song -> true
 }
 
-void Hiscore::addHiscore(unsigned score, unsigned playerid, unsigned songid, std::string const& track) {
+void Hiscore::addHiscore(unsigned score, unsigned playerid, unsigned songid, unsigned level, std::string const& track) {
 	if (track.empty()) throw std::runtime_error("No track given");
-	if (!reachedHiscore(score, songid, track)) return;
-	m_hiscore.insert(HiscoreItem(score, playerid, songid, track));
+	if (!reachedHiscore(score, songid, level, track)) return;
+	m_hiscore.insert(HiscoreItem(score, playerid, songid, level, track));
 }
 
 Hiscore::HiscoreVector Hiscore::queryHiscore(unsigned max, unsigned playerid, unsigned songid, std::string const& track) const {
 	HiscoreVector hv;
 	for (auto const& h: m_hiscore) {
 		if (playerid != unsigned(-1) && playerid != h.playerid) continue;
 		if (songid != unsigned(-1) && songid != h.songid) continue;
+		if (currentLevel() != h.level) continue;
```

Comment:
`currentLevel()` is an int while `h.level` is unsigned.
The current pipelines are nagging about this comparison:
`e/hiscore.cc:37:22:parison of integer expressions of different signedness: ‘pare`